src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -291,6 +291,30 @@ public function firstOrFail($columns = ['*'])
     }
 
     /**
+     * Traverses through a result set using a cursor.
+     *
+     * @return void
+     */
+    public function traverse()
+    {
+        $builder = $this->applyScopes();
+
+        $statement = $builder->query->fetch();
+
+        while ($row = $statement->fetch()) {
+            // On each result set, we will pass them to the callback and then let the
+            // developer take care of everything within the callback, which allows us to
+            // keep the memory low for spinning through large result sets for working.
+
+            $continue = (yield $row);
```

Comment:
First thought was allowing for a response to be sent back, but for this use case you are correct it is not needed.